indexing-service/src/main/java/org/apache/druid/indexing/common/TaskLock.java

```diff
@@ -62,4 +63,18 @@ public interface TaskLock
   boolean isRevoked();
 
   boolean conflict(LockRequest request);
+
+  /**
+   * Checks if the lock is revoked and throws a {@link DruidException} if so.
+   *
+   * @throws DruidException if the lock is revoked.
+   */
+  default void assertNotRevoked()
+  {
+    if (isRevoked()) {
+      throw DruidException.forPersona(DruidException.Persona.OPERATOR)
+          .ofCategory(DruidException.Category.RUNTIME_FAILURE)
+          .build("Lock for interval [%s] was revoked", getInterval());
```

Comment:
Nit: Druid message formatting style. Should we include lock type and version in the message too?
```suggestion
          .build("Lock for interval[%s] was revoked", getInterval());
```